File: scripts/check_fcm_tokens.js
const admin = require("firebase-admin");

admin.initializeApp();

async function checkFCMTokens() {
  const shopId = 'honC8KnKhOUG19wcYOFDTGVdKWP2';

  try {
    const userDocs = await admin.firestore()
      .collection('users')
      .where('shopId', '==', shopId)
      .get();

    console.log(`Found ${userDocs.docs.length} users in shop ${shopId}:`);

    userDocs.forEach(doc => {
      const userData = doc.data();
      const userId = doc.id;
      const fcmToken = userData.fcmToken;
      const email = userData.email;

      console.log(`User: ${userId} (${email})`);
      console.log(`  FCM Token: ${fcmToken ? fcmToken.substring(0, 50) + "..." : "NULL"}`);
      console.log(`  Token length: ${fcmToken ? fcmToken.length : 0}`);
      console.log('');
    });
  } catch (error) {
    console.error('Error:', error);
  } finally {
    admin.app().delete();
  }
}

checkFCMTokens();

File: scripts/check_fcm_tokens.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:firebase_core/firebase_core.dart';

void main() async {
  await Firebase.initializeApp();

  final shopId = 'honC8KnKhOUG19wcYOFDTGVdKWP2';

  try {
    final userDocs = await FirebaseFirestore.instance
        .collection('users')
        .where('shopId', '==', shopId)
        .get();

    print('Found ${userDocs.docs.length} users in shop $shopId:');

    for (final doc in userDocs.docs) {
      final userData = doc.data();
      final userId = doc.id;
      final fcmToken = userData['fcmToken'];
      final email = userData['email'];

      print('User: $userId ($email)');
      print('  FCM Token: ${fcmToken != null ? fcmToken.substring(0, 50) + "..." : "NULL"}');
      print('  Token length: ${fcmToken?.length ?? 0}');
      print('');
    }
  } catch (e) {
    print('Error: $e');
  }
}